var/spack/repos/builtin/packages/py-ray/package.py

```diff
@@ -36,6 +53,12 @@ class PyRay(PythonPackage):
     depends_on('py-py-spy@0.2.0:', type=('build', 'run'))
     depends_on('py-pyyaml', type=('build', 'run'))
     depends_on('py-requests', type=('build', 'run'))
-    depends_on('py-redis@3.3.2:3.4', type=('build', 'run'))
+    depends_on('py-redis@3.3.2:3.4', type=('build', 'run'), when='@:1.0.1')
+    depends_on('py-redis@3.5.0:4', type=('build', 'run'), when='@1.1.0:')
     depends_on('py-opencensus', type=('build', 'run'))
     depends_on('py-prometheus-client@0.7.1:', type=('build', 'run'))
+    depends_on('py-cloudpickle', type=('build', 'run'), when='@1.3.0:')
+    depends_on('py-pydantic@1.8.0:', type=('build', 'run'), when='@1.4.0:')
+    depends_on('py-aiosignal', type=('build', 'run'), when='@1.9.0:')
+    depends_on('py-smart-open', type=('build', 'run'), when='@1.9.0:')
+    depends_on('py-frozenlist', type=('build', 'run'), when='@1.9.0:')
```

Comment:
A lot of these dependencies are listed in `requirements.txt` but not in `setup.py`, and a lot of the dependencies listed in `setup.py` are missing in this list. I generally don't trust `requirements.txt` because it isn't used by package managers like pip (and in this case there are several different ones to choose from), I would instead only use what is declared as a dep in `setup.py` under `install_requires`.